### ISSUE TITLE
Saving?

### ISSUE THREAD

**Issue by MikeJaoudi** [orta/WibbleQuest#38]
Should we have a save state feature?

**Comment by orta** [comment]
Yes. I think we're at a point where not having it is a bit awkwards. Plus it helps that I actually have a iPhone now, so I'd like to see them both synced over iCloud. 

My previous thoughts on this had been to have a way to transform the entire object graph into a JSON data structure, and then be able to go back from that. Alas, what this means is that we'd need our own API at a lower level for finding out when properties on objects change. 

This might be possible using KeyValueWatching on the WibbleObject so that it monitors changes to its own vars. Otherwise there may be a need to have a way to say what variables you want saving on an object.
